scipy/optimize/_minimize.py

```diff
@@ -961,3 +972,42 @@ def standardize_constraints(constraints, x0, meth):
                 constraints.extend(old_constraints[1:])  # appends 1 if present
 
     return constraints
+
+
+def _optimize_result_for_equal_bounds(
+        fun, bounds, method, args=(), constraints=()
+):
+    """
+    Provides a default OptimizeResult for when a bounded minimization method
+    has (lb == ub).all().
+
+    Parameters
+    ----------
+    fun: callable
+    bounds: Bounds
+    method: str
+    constraints: Constraint
+    """
+    success = True
+    message = 'Optimization terminated successfully.'
+
+    # bounds is new-style
+    x0 = bounds.lb
+
+    if constraints:
+        constraints = standardize_constraints(constraints, x0, 'new')
+
+    for c in constraints:
+        maxcv = 0
+        pc = PreparedConstraint(c, x0)
+        violation = pc.violation(x0)
+        if np.sum(violation):
+            maxcv = max(maxcv, np.max(violation))
+            success = False
+            message = (f"The solution does not satisfy the constraints, "
+                       f"MAXCV = {maxcv}")
```

Comment:
I'm not sure if this is credible, but could the user not realize that they are passing in `lb == ub`? In this case it might be more helpful to preface this with "The lower and upper bounds are equal, yet ..."

Reply:
It might be rare, but it's certainly possible. The message for constraints could be altered to use your suggestion. Should the non-constrained versions have a similar message?  Along the lines of `'The optimization was successful, but with no varying parameters (lb==ub).all()'`.

Reply:
Yes, but I'm not sure that "optimization was successful" really needs to appear, even if `success = True`, since no optimization really occurred. How about just:

> All decision variables were fixed by `bounds`.

And if there are constraints.

> All decision variables were fixed by `bounds` at values that satisfy the constraints.